### ISSUE TITLE
Input method support?

### ISSUE THREAD

**Issue by zxygentoo** [vkbo/novelWriter#1293]
Hi @vkbo 

I'm searching for a Scrivener alternative on Linux and novelWriter seems _really_ nice.

One thing, I'm writing both in English and Chinese, and my input method seems not work with novelWriter.

Input Method: Fcitx 4.2.9.7
OS: Gentoo Linux
novelWriter: latest AppImage

According to fcitx's wiki:
```
Binary Qt application
Due to Qt 5 does not support XIM, and it only bundles ibus im module, 
you may want to set `QT_IM_MODULE=ibus` for Qt application 
that does not use your system Qt library. (It may still not work because 
certain Qt application does not even bundle any im module).
```

With some quick code search and googling, it seems to me this is the case for novelWriter? I'm not familiar with Qt5/PyQt5, so I might be wrong and this is just some setting problem on my part (Gentoo and all that...). But if it is, adding IM support will surely benefit a lot of users (eg. CJK users).

As said I'm not familiar with Qt, but did fair amount of Python and willing to help with this if necessary.

Cheers,
Alex

**Comment by vkbo** [comment]
This is beyond my experience, so I cannot answer this. Maybe @longqzh has an idea?

novelWriter runs on Qt5 only, so if Qt5 does not support this, then that is a limitation. If you're running on Linux, then you could try running novelWriter from source, in which case it will pull in system libraries. At least as a test.

**Comment by vkbo** [comment]
From the [ArchWiki](https://wiki.archlinux.org/title/Fcitx) it seems to me something can be done to solve this though, if I understand it correctly. I'm not sure if anything special is needed for it to work in an AppImage though.

**Comment by zxygentoo** [comment]
Thanks for the response, will try running from source and see if this can be fixed using system's Qt. (I'm running a pretty lean build and have no Qt installed in my system, so I tried the AppImage version first.)

**Comment by vkbo** [comment]
Great! Let me know if there is anything that needs to be done in the source to help with this.

**Comment by vkbo** [comment]
Since there has been no more updates on this for a couple of months, I'm closing this. Let me know if there is anything else.

**Comment by zxygentoo** [comment]
Sorry for taking so long to get back to you. Finally spent some time to figure it out. On my system (which is a pretty minimal Gentoo), the following fix things for me. Maybe it will be useful for some one else.

1. Install fcitx-qt5 (I just emerged the one from the official Gentoo portage, and I believe most _normal_ distro will install this along with fcitx).
2. Clone this repo (I switched to v2.4 just in case)
3. Make a venv, activate it, install deps from requirements.txt
5. copy fcitx-qt5 plugin to venv `cp /usr/lib64/qt5/plugins/platforminputcontexts/libfcitxplatforminputcontextplugin.so venv/lib64/python3.11/site-packages/PyQt5/Qt5/plugins/platforminputcontexts/libfcitxplatforminputcontextplugin.so`
6. run novelWriter `python3 novelWriter.py`

I have no idea if this is easily fixable for AppImage. But, fcitx is indeed pretty popular at least for Chinese.

---
Starting a new writing project recently, will try to make that scrivener-to-novelwriter migration. As always, thanks for this nice software.
